### PR TITLE
Update installation scripts

### PR DESCRIPTION
**Pull request by sk593**
# Description

Update radius installation scripts to pull from GitHub Releases 

Screenshot 2023-10-17 at 1 31 49 PM

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### Summary
🚀🐛🎨

This pull request enhances the `deploy/install.ps1` and `deploy/install.sh` scripts to make them more robust, user-friendly, and consistent for installing the `rad` CLI tool from GitHub releases.

> _We are the masters of the `rad` CLI_
> _We download and install it with ease_
> _No matter if you use Windows or Linux_
> _We handle errors and output with finesse_

### Walkthrough
*  Use GitHub API to get latest release of rad CLI binary for Windows and Linux/Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL32-R74), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL75-R156), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL136-L135), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L103-R107), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L118-R125), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L210-R232))
*  Simplify and update logic for copying rad CLI binary to /usr/local/bin for Linux and Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L22-R22), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L43-R48), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L118-R125), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177))
*  Add support for darwin-arm64 and darwin-amd64 OS-ARCH combinations ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L56-R54), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L118-R125))
*  Print version string of installed rad CLI and rad-bicep component ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL136-L135), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L177-R188))
*  Remove extra newline in output message when detecting existing rad CLI version ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L95-R93))
*  Remove redundant verification of downloaded executable ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177))
*  Remove unused OS mapping from Darwin to macos ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L43-R48))
*  Add v prefix to release tag name if specified as argument ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L210-R232))
*  Throw error if specified release version does not exist ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL75-R156), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177))
*  Define helper functions for parsing release JSON and getting asset URL and name for Windows ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL32-R74), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL75-R156))
*  Handle authentication header for GitHub requests and progress preference for PowerShell for Windows ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL75-R156))
*  Create install directory if it does not exist for Linux and Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177))
*  Create temp directory with specific prefix for Linux and Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L118-R125))
*  Check HTTP status code of download URL before downloading for Linux and Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L118-R125))
*  Remove old rad CLI binary if it exists for Windows and Linux/Darwin ([link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-506d53c740958f1f7b269e6b80b68300aeb53417fccb0297240700f1d84811bfL136-L135), [link](https://github.com/radius-project/radius/pull/6495/files?diff=unified&w=0#diff-ed1191c237ea4ba8f2011568f4ed0208c8d0ca36cbde1285828a61b4ff2128b4L148-R177))

### CODE REVIEW

File: deploy/install.sh [hunk 1]
```diff
@@ -198,7 +209,7 @@ cleanup() {
 }
 
 installCompleted() {
-    echo -e "\nTo get started with Radius, please visit https://radapp.dev/getting-started/"
+    echo "\nTo get started with Radius, please visit https://radapp.dev/getting-started/"
```
Comment (AaronCrawfis) [inline deploy/install.sh:212]:
```suggestion
    echo "\nTo get started with Radius, please visit https://radapp.io/getting-started/"
```

**Review by lakshmimsft — DISMISSED** [review]

File: deploy/install.sh (hunk 1 above)
Comment (sk593) [reply deploy/install.sh]: Updated

File: deploy/install.sh
Comment (AaronCrawfis) [inline deploy/install.sh:1]: Getting this error when trying to run on WSL:

```
Getting the latest Radius CLI...
Your system is linux_x64
/usr/local/bin/rad: line 1: Not: command not found
\nRadius CLI is detected. Current version:
Reinstalling Radius CLI - /usr/local/bin/rad...\n
Installing v0.26.0 Radius CLI...
Downloading https://github.com/radius-project/radius/releases/download/v0.26.0/rad_linux_x64 ...
[sudo] password for aacrawfi:
rad installed into /usr/local/bin successfully.
Installing rad-bicep ("rad bicep download")...
/usr/local/bin/rad: line 1: Not: command not found
Failed to install rad-bicep
Failed to install Radius CLI
For support, go to https://github.com/radius-project/radius
```

Comment (AaronCrawfis) [reply deploy/install.sh:1]:
```
$ cat rad
Not Found
```

Comment (sk593) [reply deploy/install.sh:1]: Looking

Comment (sk593) [reply deploy/install.sh:1]: Updated, this was pulling the wrong arch

Comment (AaronCrawfis) [reply deploy/install.sh:1]: Working, thanks!

Comment (AaronCrawfis) [inline deploy/install.sh:1]: Getting some formatting issues:

```
aacrawfi@Crawfis-Z4:/usr/local/bin$ wget -q https://raw.githubusercontent.com/radius-project/radius/6c70f0578d53c6a9872481f985785e715afc3a0b/deploy/install.sh -O - | /bin/bash
Getting the latest Radius CLI...
Your system is linux_amd64
/usr/local/bin/rad: line 1: Not: command not found
\nRadius CLI is detected. Current version:
Reinstalling Radius CLI - /usr/local/bin/rad...\n
Installing v0.26.0 Radius CLI...
Downloading https://github.com/radius-project/radius/releases/download/v0.26.0/rad_linux_amd64 ...
[sudo] password for aacrawfi:
rad installed into /usr/local/bin successfully.
Installing rad-bicep ("rad bicep download")...
Downloading Bicep for channel 0.26...
rad-bicep installed successfully
\nTo get started with Radius, please visit https://radapp.io/getting-started/
aacrawfi@Crawfis-Z4:/usr/local/bin$ rad version
RELEASE   VERSION   BICEP     COMMIT
0.26.0    v0.26.0   0.11.90   74038ee1ae2ca6496ad9fda81998677683fc95d6
```

Specifically:

- `/usr/local/bin/rad: line 1: Not: command not found` should not be there
- `\nRadius CLI is detected. Current version:` shows the `\n` character
- `\nTo get started with Radius, please visit https://radapp.io/getting-started/` shows the `\n`

Comment (sk593) [reply deploy/install.sh:1]: Looking, not sure why the new line characters are showing up

Comment (lakshmimsft) [reply deploy/install.sh:1]: fyi , no new line chars on M1:
image

Comment (AaronCrawfis) [reply deploy/install.sh:1]: I think you need the `-e` in `echo` to interpret backslashes

Comment (sk593) [reply deploy/install.sh:1]: Yea, not needed for Mac but I think for other arch's it might be

**Review by willdavsmith — DISMISSED** [review]

File: deploy/install.sh
Comment (sk593) [reply deploy/install.sh:1]: Updated the formatting, synced offline and the other issue seems to be from a previously broken rad download

File: deploy/install.sh [hunk 2]
```diff
@@ -115,55 +112,69 @@ getLatestRelease() {
...
+    echo "Downloading $DOWNLOAD_URL ..."
     if [ "$RADIUS_HTTP_REQUEST_CLI" == "curl" ]; then
         curl -SsL "$DOWNLOAD_URL" -o "$ARTIFACT_TMP_FILE"
     else
         wget -q -O "$ARTIFACT_TMP_FILE" "$DOWNLOAD_URL"
     fi
 
     if [ ! -f "$ARTIFACT_TMP_FILE" ]; then
-        echo "failed to download ${DOWNLOAD_URL}..."
+        echo "failed to download $DOWNLOAD_URL ..."
```
Comment (AaronCrawfis) [inline deploy/install.sh:131]:
```suggestion
        echo "failed to download ${DOWNLOAD_URL}..."
```

There was an extra period between the URL and the periods

File: deploy/install.sh [hunk 3]
```diff
@@ -115,55 +112,69 @@ getLatestRelease() {
...
-    else
-        if ! wget --spider "$DOWNLOAD_URL" 2>/dev/null; then
-            echo "ERROR: The specified release version: $LATEST_RELEASE_TAG does not exist."
-            exit 1
-        fi
-    fi
-
-
-    echo "Downloading ${DOWNLOAD_URL}..."
+    echo "Downloading $DOWNLOAD_URL ..."
```
Comment (AaronCrawfis) [inline deploy/install.sh:123]:
```suggestion
    echo "Downloading ${DOWNLOAD_URL}..."
```

File: deploy/install.sh (hunk 2 above)
Comment (sk593) [reply deploy/install.sh]: Updated

File: deploy/install.sh (hunk 3 above)
Comment (sk593) [reply deploy/install.sh]: Updated

File: deploy/install.sh [hunk 4]
```diff
@@ -115,55 +112,69 @@ getLatestRelease() {
...
+    if [ -f "$RADIUS_CLI_FILE" ]; then
+        runAsRoot rm "$RADIUS_CLI_FILE"
+    fi
+    chmod o+x $tmp_root_radius_cli
+    mkdir -p $RADIUS_INSTALL_DIR
+    runAsRoot cp "$tmp_root_radius_cli" "$RADIUS_INSTALL_DIR"
+    runAsRoot mv "${RADIUS_INSTALL_DIR}/${RADIUS_CLI_ARTIFACT}" "${RADIUS_INSTALL_DIR}/${RADIUS_CLI_FILENAME}"
 
     if [ -f "$RADIUS_CLI_FILE" ]; then
         echo "$RADIUS_CLI_FILENAME installed into $RADIUS_INSTALL_DIR successfully."
```
Comment (AaronCrawfis) [inline deploy/install.sh:176]:
```suggestion
        echo "$RADIUS_CLI_FILENAME installed into $RADIUS_INSTALL_DIR successfully"
```

File: deploy/install.sh [hunk 5]
```diff
@@ -198,7 +209,7 @@ cleanup() {
 }
 
 installCompleted() {
-    echo -e "\nTo get started with Radius, please visit https://radapp.dev/getting-started/"
+    echo -e "\nTo get started with Radius, please visit https://radapp.io/getting-started/"
```
Comment (AaronCrawfis) [inline deploy/install.sh:212]:
```suggestion
    echo -e "\nTo get started with Radius, please visit https://docs.radapp.io/getting-started/"
```

**Review by AaronCrawfis — APPROVED** [review]

**Review by vishwahiremat — APPROVED** [review]

**Review by shalabhms — APPROVED** [review]

**Pull request merged**